src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -732,10 +732,26 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                 throw new InvalidOperationException("An 'id' must be specified in the host configuration.");
             }
 
-            JToken watchFiles = (JToken)config["watchFiles"];
-            if (watchFiles != null && watchFiles.Type == JTokenType.Boolean)
+            JToken fileWatchingEnabled = (JToken)config["fileWatchingEnabled"];
+            if (fileWatchingEnabled != null && fileWatchingEnabled.Type == JTokenType.Boolean)
             {
-                scriptConfig.FileWatchingEnabled = (bool)watchFiles;
+                scriptConfig.FileWatchingEnabled = (bool)fileWatchingEnabled;
+            }
+
+            // Configure the set of watched directories, adding the standard built in
+            // set to any the user may have specified
+            if (scriptConfig.WatchDirectories == null)
+            {
+                scriptConfig.WatchDirectories = new Collection<string>();
+            }
+            scriptConfig.WatchDirectories.Add("node_modules");
+            JToken watchDirectories = config["watchDirectories"];
+            if (watchDirectories != null && watchDirectories.Type == JTokenType.Array)
```

Comment:
Logged https://github.com/Azure/azure-webjobs-sdk-script/issues/614 to take care of this across the board. It will take a bit of time.